src/providers/VoiceFocusProvider/index.tsx

```diff
@@ -0,0 +1,133 @@
+                    const transformer = await getVoiceFocusDeviceTransformer();
+                    if(transformer && transformer.isSupported()){
+                        setSupportsVoiceFocus(true);
+                        return;
+                    }
+                }
+            } catch (e) {
+                console.warn(`Amazon Voice Focus is not supported: ${e.message}`);
+            }
+            setSupportsVoiceFocus(false);
+        };
+        initVoiceFocus();
+    }, []);
+
+    useEffect(() => {
+        if(supportsVoiceFocus) {
+            console.log('Amazon Voice Focus is supported.');
+        } else {
+            console.warn('Amazon Voice Focus is not supported.');
+        }
+    }, [supportsVoiceFocus]);
+
+    async function getVoiceFocusDeviceTransformer(): Promise<VoiceFocusDeviceTransformer> {
+        if (voiceFocusTransformer) return voiceFocusTransformer;
+        const transformer = await VoiceFocusDeviceTransformer.create(spec, options);
+        setVoiceFocusTransformer(transformer);
+        return transformer;
+    }
+
+    const transformToVoiceFocus = useCallback(async (device: Device): Promise<Device | VoiceFocusTransformDevice> => {
+        let intrinsic = device;
+        if (device instanceof VoiceFocusTransformDevice) {
+            intrinsic = device.getInnerDevice();
+        }
+        return await audioInputSelectionWithOptionalVoiceFocus(intrinsic);
+    }, []);
+
+    async function audioInputSelectionWithOptionalVoiceFocus(device: Device): Promise<Device | VoiceFocusTransformDevice> {
+        if(!voiceFocusDevice) {
+            return await createVoiceFocusDevice(device);
+        }
+        const vf = await voiceFocusDevice.chooseNewInnerDevice(device);
+        setVoiceFocusDevice(vf);
+        return vf;
+    }
+
+    async function createVoiceFocusDevice(inner: Device): Promise<Device | VoiceFocusTransformDevice> {
+        if(!supportsVoiceFocus) {
+            return inner;
+        } 
+        try {
+            const transformer = await getVoiceFocusDeviceTransformer();
+            const device = await transformer?.createTransformDevice(inner);
+            if(device) {
+                setVoiceFocusDevice(device);
+                return device;
+            }
+        } catch (e) {
+            console.warn(`Amazon Voice Focus is not supported: ${e.message}`);
+        }
+        return inner;
+    }
+
+    const value: VoiceFocusState = useMemo(
+        () => ({
+            supportsVoiceFocus,
+            transformToVoiceFocus
+        }),
+        [supportsVoiceFocus, transformToVoiceFocus]
+    );
+
+    return (
+        <VoiceFocusContext.Provider value={value}>
+            {children}
+        </VoiceFocusContext.Provider>
+    );
+}
+
+const useVoiceFocus = (): VoiceFocusState => {
+    const context = useContext(VoiceFocusContext);
+
+    if(!context) {
+        throw new Error(
+            'useVoiceFocus must be used within VoiceFocusProvider'
+        );
+    }
```

Comment:
Huh, I thought `useContext` did this automatically. Could you double-check by removing the provider from your test JSX?

Reply:
Checked that `useContext` does not throw this error automatically. would just keep it